src/Misc/PerformanceViewer/perfCollectionType.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * Defines the predefined collection types used in the performance viewer.
+ */
+export enum PerfCollectionType {

```

Comment:
This will probably not scale enough :-) and we ll be hard to use for custom inputs ?

Reply:
Well this mapping is only for the predefined collection types! Custom inputs will have their own strategies defined separately from the functions I've created here. The reason for the "closed-ness" is because there isn't really a need to check if we have enabled a custom input, because it wouldn't exist if we had not enabled it. 

we will probably create a separate record for "general strategies" if I find that the logic is shared enough for it (I suspect it will have some shared aspects), but populate the same "strategies" map if we find we have registered some custom events.

Reply:
Discussed offline. The concern is more so if we expand our use cases. Because of this I have opted to use enum string instead. It has allowed the code to be cleaned up a bit as well! 😄